src/Testing/TypeInferenceTestCase.php

```diff
@@ -126,12 +133,73 @@ public function assertFileAsserts(
 		}
 	}
 
+	public static function checkOnlyIfAnnotation(string $file): bool
```

Comment:
might be helpful to move this logic into a dedicated class and do some unit testing on the parser

Reply:
It makes sense, even though testing the tests goes into dangerous territory 😆 
Just kidding.. I think it makes sense, I can move it to src/Testing and make a simple test case for it.

Reply:
On a second thought we are already in src/Testing I realized, so maybe we should make it a class only if we ever need it in another class, so we don't have a weird invocation of an unrelated class's method.. but I'm OK either way.

Reply:
my comment was misleading. 

I was about extracting the parsing logic of this method, but leave the method here to be available for other users of TypeInferenceTestCase (its a public API class and used in phpstan extension testing outside this repo)

Reply:
side note: should this method be private?

Reply:
I made it protected, otherwise I cannot test it?